pkg/agent/controller/networkpolicy/fqdn.go

```diff
@@ -809,6 +809,13 @@ func (f *fqdnController) sendDNSPacketout(pktIn *ofctrl.PacketIn) error {
 		}
 	}
 	if prot == protocol.Type_UDP {
+		inPort := uint32(config.HostGatewayOFPort)
+		// Use the actual in_port number in the packetIn message to set the in_port field of the generated packetOut message.
+		matches := pktIn.GetMatches()
+		ofPortField := matches.GetMatchByName("OXM_OF_IN_PORT")
+		if ofPortField != nil {
+			inPort = ofPortField.GetValue().(uint32)
+		}
```

Comment:
This should work, but I think we'd better expose an interface/method to modules to GetHostGatewayOFPort/GetXXXOFPort

Reply:
Do you mean add the method in ofnet, or in Antrea code?

Reply:
In Antrea code. I mean we should provide common method to get these OFPorts, like I commented below.